### PR TITLE
Updated logging, implemented batch processing and partitioning.

### PR DESCRIPTION
**Pull request by devendra-bhatt-a10**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM ubuntu:latest
-RUN apt-get update -y
+RUN apt-get update
+RUN apt-get upgrade -y
```
Comment (SwapnilKashidA10) [inline Dockerfile:3]: does upgrade is needed, as it will increase size of image

File: acos_exporter.py [hunk 2]
```diff
@@ -8,12 +8,15 @@
...
 from prometheus_client import Gauge
 import logging
+from logging.handlers import RotatingFileHandler
 
 UNDERSCORE = "_"
 SLASH = "/"
 HYPHEN = "-"
 PLUS = "+"
 
+log_file_size = 5*1024*1024
```
Comment (SwapnilKashidA10) [inline acos_exporter.py:18]: Constant in Capital letters

File: acos_exporter.py [hunk 3]
```diff
@@ -81,88 +91,149 @@ def getauth(host):
             logger.error("password not provided.")
 
         payload = {'Credentials': {'username': uname, 'password': pwd}}
-        auth = json.loads(
-            requests.post("https://{host}/axapi/v3/auth".format(host=host), json=payload, verify=False).content.decode(
-                'UTF-8'))
+        try:
+            auth = json.loads(requests.post("https://{host}/axapi/v3/auth".format(host=host), json=payload,
+                                            verify=False, timeout=5).content.decode('UTF-8'))
```
Comment (SwapnilKashidA10) [inline acos_exporter.py:96]: Time out seconds should be constant

Comment (SwapnilKashidA10) [reply acos_exporter.py:96]: In exception too

File: acos_exporter.py [hunk 4]
```diff
@@ -81,88 +91,149 @@ def getauth(host):
...
+
         if 'authresponse' not in auth:
             logger.error("Host credentials are not correct")
             return ''
         return 'A10 ' + auth['authresponse']['signature']
 
 
+def get_stats(body, endpoint, host_ip, headers):
+    batch_endpoint = "/batch-get"
+    logger.info("Uri - " + endpoint + batch_endpoint)
```
Comment (SwapnilKashidA10) [inline acos_exporter.py:109]: Logging of list of end points is also needed

File: acos_exporter.py [hunk 5]
```diff
@@ -81,88 +91,149 @@ def getauth(host):
...
+    api_names = request.args.getlist("api_name")
+    partition = request.args.get("partition", "shared")
+    res = []
+
+    # Generate batch API body with list of APIs provided by user.
+    body = {
+        "batch-get-list": list()
+    }
+    for api_endpoint in api_endpoints:
+        body["batch-get-list"].append({"uri": "/axapi/v3" + api_endpoint + "/stats"})
```
Comment (SwapnilKashidA10) [inline acos_exporter.py:157]: Body can be added in get stats method

File: acos_exporter.py (hunk 4 above)
Comment (devendra-bhatt-a10) [reply acos_exporter.py:109]: It is done initially itself after parsing query params.

File: Dockerfile (hunk 1 above)
Comment (devendra-bhatt-a10) [reply Dockerfile]: OKay removed it

File: acos_exporter.py [hunk 6]
```diff
@@ -81,88 +92,149 @@ def getauth(host):
...
-    for name in global_api_collection[api]:
-        res.append(prometheus_client.generate_latest(global_api_collection[api][name]))
+    # Changing Partition if provided.
+    logger.info("partition - " + str(partition))
+    if "shared" not in partition:
+        change_partition(partition, endpoint, headers)
+        response = get_stats(api_endpoints, endpoint, host_ip, headers)
+        change_partition("shared", endpoint, headers)
+    else:
+        response = get_stats(body, endpoint, host_ip, headers)
```
Comment (SwapnilKashidA10) [inline acos_exporter.py:190]: Body should be related with api endpoints

File: acos_exporter.py [hunk 7]
```diff
@@ -81,88 +92,149 @@ def getauth(host):
...
-            global_stats[key].labels(api).set(stats[org_key])
-
-    global_api_collection[api] = current_api_stats
-    res = []
-    for name in global_api_collection[api]:
-        res.append(prometheus_client.generate_latest(global_api_collection[api][name]))
+    # Changing Partition if provided.
+    logger.info("partition - " + str(partition))
+    if "shared" not in partition:
+        change_partition(partition, endpoint, headers)
```
Comment (SwapnilKashidA10) [inline acos_exporter.py:186]: Exception handling in api call

Comment (SwapnilKashidA10) [reply acos_exporter.py:186]: Like auth api

File: acos_exporter.py (hunk 6 above)
Comment (SwapnilKashidA10) [reply acos_exporter.py:190]: Replaced

**Pull request merged**